src/server/pkg/storage/fileset/index/reader.go

```diff
@@ -145,13 +152,37 @@ func (r *Reader) next(level int) (*Header, error) {
 			return nil, err
 		}
 		// Skip to the starting header.
-		if fullHdr.Idx.Range.LastPath < r.prefix {
+		if !r.atStart(fullHdr.Idx.Range.LastPath) {
 			continue
 		}
 		return fullHdr, nil
 	}
 }
 
+func (r *Reader) atStart(name string) bool {
+	if r.filter == nil {
+		return true
+	}
+	if r.filter.pathRange != nil {
+		return name >= r.filter.pathRange.Lower
+	}
+	return name >= r.filter.prefix
+}
+
+func (r *Reader) atEnd(name string) bool {
+	if r.filter == nil {
+		return false
+	}
+	if r.filter.pathRange != nil {
+		return name > r.filter.pathRange.Upper
+	}
+	// Name is past a prefix when the first len(prefix) bytes are greater than the prefix
+	// (use len(name) bytes for comparison when len(name) < len(prefix)).
+	// Cannot use a simple greater than check (for paths a, ab, abc, and b, prefix a should return a, ab, abc).
+	cmpSize := int64(math.Min(float64(len(name)), float64(len(r.filter.prefix))))
```

Comment:
Converting to floats makes me a little uneasy here (even though we would have to have some massive names for it to not get the right answer here.) Might as well copy the int min function we have all over the place (or put it in a common place.)

Reply:
I found a math utility package that has a lot of the common math functions that are not a part of the standard math package. It would be a good idea for engineers to look at it first before adding a math function that is not provided by the standard math package.